x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/recall_from_connectors.ts

```diff
@@ -96,17 +99,25 @@ export async function recallFromConnectors({
 }
 
 async function getConnectorIndices(
-  esClient: { asCurrentUser: ElasticsearchClient },
-  uiSettingsClient: IUiSettingsClient
+  esClient: { asCurrentUser: ElasticsearchClient; asInternalUser: ElasticsearchClient },
+  uiSettingsClient: IUiSettingsClient,
+  logger: Logger
 ) {
   // improve performance by running this in parallel with the `uiSettingsClient` request
-  const responsePromise = esClient.asCurrentUser.transport.request({
-    method: 'GET',
-    path: '_connector',
-    querystring: {
-      filter_path: 'results.index_name',
-    },
-  });
+  const responsePromise = esClient.asInternalUser.transport
+    .request<{
+      results?: Array<{ index_name: string }>;
+    }>({
+      method: 'GET',
+      path: '_connector',
+      querystring: {
+        filter_path: 'results.index_name',
+      },
+    })
+    .catch((e) => {
+      logger.warn(`Failed to fetch connector indices due to ${e.message}`);
+      return { results: [] };
+    });
```

Comment:
Note-worthy change: this catches errors and logs them instead of bubbling them up. This prevents the server from crashing during development